localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -247,6 +247,9 @@ def get_metric_data_stat(
                 dimension_filter += "AND dimensions LIKE ? "
                 data = data + (f"%{dimension.get('Name')}={dimension.get('Value','')}%",)
 
+            if not dimensions:
+                dimension_filter = "AND dimensions is null "
+
```

Comment:
It would be interesting to see if you can actually filter for data without dimensions this way (does something like that even exist?), and if AWS handles this case as implied by this change. Then again, I guess we have to finish our fringe case testing at some point. So I will leave this to @steffyP if we need another test case here or not (or if it is easy we could just write one)